cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizationEnforcementModule.java

```diff
@@ -48,7 +48,16 @@ protected void configure() {
           .in(Scopes.SINGLETON);
         // bind AuthorizationEnforcer to AuthorizationEnforcementService
         bind(AuthorizationEnforcer.class).to(AuthorizationEnforcementService.class).in(Scopes.SINGLETON);
+
+        bind(PrivilegesFetcherProxyService.class).to(DefaultPrivilegesFetcherProxyService.class)
+          .in(Scopes.SINGLETON);
         bind(PrivilegesFetcher.class).to(AuthorizerAsPrivilegesFetcher.class).in(Scopes.SINGLETON);
+        bind(PrivilegesFetcher.class)
+          .annotatedWith(Names.named(PRIVILEGES_FETCHER_PROXY_CACHE))
+          .to(PrivilegesFetcherProxyService.class);
```

Comment:
Yes. There are two caches. An enforcement cache is DefaultAuthorizationEnforcementCache. This is the one that's started used in master as well as system and program containers for enforcement. Internally, it uses a PrivilegesFetcher to update its cache. When the enforcement cache runs in the master, the PrivilegesFetcher used is the configured Authorizer, that talks directly to Sentry. When enforcement cache runs inside some other container, the PrivilegesFetcher is bound to a RemotePrivilegesFetcher, which makes an HTTP call to master, to fetch privileges. The Handler serving this HTTP Request uses a proxy cache - PrivilegeFetcherProxyService. This guy runs inside the master and keeps updating privileges, and serves the HTTP Requests to update privileges from program and system containers.